scripts/naming_conventions_yaml_parser.py

```diff
@@ -0,0 +1,117 @@
+'''
+This is an experimental prototype that generates naming convention checkers from a .yaml file. Currently the templates
+used to identify the object being checked are stored in /afs/cern.ch/user/d/daho/yaml/templates.
+
+David Ho
+August 2016
+'''
+
+import yaml
+import argparse
+import os
+import datetime
+import re
+
+class ConventionReader(object):
+    """Reader class"""
+    def __init__(self, yamlfile):
+        super(ConventionReader, self).__init__()
+        self.yamlfile = yamlfile
+        self.rules = {}
+        self.project_name = ""
+        self.known_clang_types= {
+            "Function": "FunctionDecl",
+            "Type": "TypeDecl",
+            "DataMember": "FieldDecl",
+            "Const": "VarDecl",
+            "Enumerator": "EnumDecl",
+            "Namespace": "NamespaceDecl"
+        }
+        self.template_directory = "/afs/cern.ch/user/d/daho/yaml/templates/identifiers"
+        self.identifiers = {}
+        for file in os.listdir(self.template_directory):
+            file_name, file_extension = os.path.splitext(file)
+            if file_extension == ".template":
+                template_name = os.path.splitext(file_name)[0]
+                with open(os.path.join(self.template_directory, file), "r") as fobj:
+                    self.identifiers[template_name] = fobj.read()
+        with open("/afs/cern.ch/user/d/daho/yaml/templates/DefaultIdentifier.template", "r") as fobj:
+            self.default_template = fobj.read()
+
+    def check(self, name, regex):
+        # is the regex valid?
+        try:
+            re.compile(regex)
+        except:
+            print("Invalid regex for '{name}'; checker not written".format(name=name))
+            return False
+        # is there a known Clang type associated with the name?
+        try:
+            self.known_clang_types[name]
+        except KeyError:
+            print("Unable to find Clang type for '{name}'; checker not written".format(name=name))
+            return False
+        return True
+
+    def read(self):
+        with open(self.yamlfile, 'r') as yamlfile:
+            content = yaml.load(yamlfile)
+        if "project" in content.keys():
+            self.project_name = content["project"]["name"]
+            self.project_author = content["project"]["author"]
+        if "conventions" in content.keys():
+            conventions = content["conventions"]
+            for convention_name in conventions.keys():
+                name = conventions[convention_name]["name"]
```

Comment:
Here we should check if the keys name, regex and description are all present. I think we want to make all of them mandatory and not generate if one or more are missing.